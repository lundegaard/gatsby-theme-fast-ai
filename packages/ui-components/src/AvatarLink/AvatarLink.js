import React from 'react';
import PropTypes from 'prop-types';

import Avatar from '../Avatar';
import Link from '../Link';

const AvatarLink = ({ sx, ...rest }) => (
	<Avatar
		as={Link}
		sx={{
			textDecoration: 'none',

			'&, :visited': {
				opacity: 1,
				transition:
					'background-color 200ms ease-in-out, border 200ms ease-in-out, color 200ms ease-in-out',
			},
			':hover,:focus': {
				backgroundColor: 'transparent',
				borderColor: 'white',
				color: 'contrast',
			},
			...sx,
		}}
		{...rest}
	/>
);

AvatarLink.propTypes = {
	sx: PropTypes.object,
};

export default AvatarLink;
