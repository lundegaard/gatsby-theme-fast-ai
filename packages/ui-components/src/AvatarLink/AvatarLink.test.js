import React from 'react';
import { shallow } from 'enzyme';

import AvatarLink from './AvatarLink';

describe('<AvatarLink />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<AvatarLink {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
