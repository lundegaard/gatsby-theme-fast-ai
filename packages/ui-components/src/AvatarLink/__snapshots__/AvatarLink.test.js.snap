// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarLink /> matches snapshot when shallowly rendered 1`] = `
<Avatar
  as={[Function]}
  sx={
    Object {
      "&, :visited": Object {
        "opacity": 1,
        "transition": "background-color 200ms ease-in-out, border 200ms ease-in-out, color 200ms ease-in-out",
      },
      ":hover,:focus": Object {
        "backgroundColor": "transparent",
        "borderColor": "white",
        "color": "contrast",
      },
      "textDecoration": "none",
    }
  }
/>
`;
